prowler/providers/aws/services/awslambda/awslambda_service.py

```diff
@@ -63,27 +64,42 @@ def __list_functions__(self, regional_client):
 
     def __get_function_code__(self):
         logger.info("Lambda - Getting Function Code...")
-        try:
-            for function in self.functions.values():
-                regional_client = self.regional_clients[function.region]
-                function_information = regional_client.get_function(
-                    FunctionName=function.name
-                )
-                if "Location" in function_information["Code"]:
-                    code_location_uri = function_information["Code"]["Location"]
-                    raw_code_zip = requests.get(code_location_uri).content
-                    function_code = LambdaCode(
-                        location=code_location_uri,
-                        code_zip=zipfile.ZipFile(io.BytesIO(raw_code_zip)),
+        # Use a thread pool handle the queueing and execution of the __fetch_function_code tasks, up to max_workers tasks concurrently.
+        with ThreadPoolExecutor(max_workers=10) as executor:
+            futures = {
+                executor.submit(self.__fetch_function_code, function): function
+                for function in self.functions.values()
+            }
+
+            for future in as_completed(futures):
+                function = futures[future]
+                try:
+                    function_code = future.result()
+                    if function_code:
+                        yield function, function_code
+                except Exception as error:
+                    logger.error(
```

Comment:
Can you use the same logger format as the other logs in the class?

Reply:
Will do